client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -178,13 +178,16 @@ const PatternAssembler = ( {
 
 	const trackEventContinue = () => {
 		const patterns = getPatterns();
+		const categories = Array.from( new Set( patterns.map( ( { category } ) => category?.name ) ) );
```

Comment:
Since `category?` is (technically) nullable, it includes `undefined` if we have an object like the one below.

```js
const patternsTest = [
  {category: {name: 'a'}},
  {},
  {category: {name: 'b'}},
  {category: {name: 'c'}}
] 
Array.from( new Set( patternsTest.map( ( { category } ) => category?.name ) ) );

// output: ['a', undefined, 'b', 'c']
// So categories.join( ',' ) will be 'a,,b,c'
```

Is it possible to get the data structure above? (no `category` property in `pattern`)
If it can't actually happen, I don't see the problem!

Reply:
> Is it possible to get the data structure above? (no category property in pattern)

That won't be possible because all patterns have a category. You can find how the patterns are grouped by category [here](https://github.com/Automattic/wp-calypso/blob/abaa8eee9298e2c39623bb0e50c5cee4e306f212/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-patterns-map-by-category.ts).

Reply:
Thank you! Then we can remove `?`, I believe.

```suggestion
		const categories = Array.from( new Set( patterns.map( ( { category } ) => category.name ) ) );
```